### ISSUE TITLE
The '.url' files can also be selected despite of allowedExtensions: ['zip']

### ISSUE THREAD

**Issue by Gyun-Heo** [miguelpruivo/flutter_file_picker#1286]
**Describe the bug**
The '.url' files can also be selected despite of allowedExtensions: ['zip']

**Platform**
- [O] Desktop

**Platform OS version**
Windows 10 Pro
Version : 21H2
OS Build : 19044.2965
Windows Feature Experience Pack 1000.19041.1000.0

**How are you picking?**

    FilePickerResult? result = await FilePicker.platform.pickFiles(
        dialogTitle: '검사 압축 파일 열기',
        allowMultiple: false,
        type: FileType.custom,
        allowedExtensions: ['zip'],
        lockParentWindow: true);

**Screenshots and/or video**

Example.url, FlutterDart!.url
![image](https://github.com/miguelpruivo/flutter_file_picker/assets/126963509/0c52b6ec-0352-4d60-ba60-977aaf82cec6)

**Flutter Version details**

[√] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.19044.2965], locale ko-KR)
    • Flutter version 3.10.0 on channel stable at C:\tools\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (9 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at C:\Users\Gyun\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)
    • All Android licenses accepted.

[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.5.0)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.5.33414.496
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2022.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-9586694)

[√] VS Code, 64-bit edition (version 1.78.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.64.0

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.2965]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 113.0.1774.42

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

**Comment by Gyun-Heo** [comment]
I solved this problem by adding the following code.

FilePickerResult? result = await FilePicker.platform.pickFiles();
String? fullPathFileName = result.files.single.path;

if (!fullPathFileName.toLowerCase().endsWith('.zip')) {
    return false;
  }

**Comment by philenius** [comment]
Hi @Gyun-Heo thank you for reporting this issue and your detailed description. I was able to reproduce this problem and I couldn't figure out a way to fix it :/

I only found [this question on StackOverflow](https://stackoverflow.com/questions/56060231/openfiledialog-keep-showing-lnk-files-wpf) which references an [old post on MSDN from 2015](https://social.msdn.microsoft.com/Forums/sqlserver/en-US/711ef1d3-0e5f-4836-804a-78402ea8bccc/why-does-openfiledialog-contain-shortcuts-to-noncsv-files?forum=vbgeneral).

Also, I checked the [OPENFILENAMEW documentation](https://learn.microsoft.com/en-us/windows/win32/api/commdlg/ns-commdlg-openfilenamew) for any flags that would allow us to hide shortcut/link files. There's only a flag named `OFN_NODEREFERENCELINKS` which provides the following functionality:
> Directs the dialog box to return the path and file name of the selected shortcut (.LNK) file. If this value is not specified, the dialog 
box returns the path and file name of the file referenced by the shortcut. 
Which makes the situation even worse as it returns the path to the shortcut file itself:
![no-dereference-shortcut](https://github.com/miguelpruivo/flutter_file_picker/assets/9049899/445944ab-977e-47c3-9396-94d1469204a5)


I'm sorry but it seems that neither `OPENFILENAME` nor the `OpenFileDialog` API (the Windows API that we're using to open the file picker dialog) can filter out shortcuts / links. You will have to stick to your workaround where you check the file extension after the user picked a file.
